daemon/src/model/cfd.rs

```diff
@@ -2373,6 +2390,219 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn given_collab_settlement_then_cannot_start_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
+
+        let cfd = Cfd::taker_long()
+            .with_quantity(quantity)
+            .with_opening_price(opening_price)
+            .with_leverage(leverage)
+            .dummy_open(dummy_event_id());
+
+        let _ = assert!(cfd.start_rollover().is_ok(), "Could not start rollover");
+
+        let (cfd, _, _, _) = cfd
+            .with_lock(taker_keys, maker_keys)
+            .dummy_collab_settlement_taker(opening_price);
+
+        let err = cfd.start_rollover().unwrap_err();
+        let no_rollover_reason = err.downcast::<NoRolloverReason>().unwrap();
+        assert_eq!(no_rollover_reason, NoRolloverReason::Closed);
+    }
+
+    /// Cover scenario where trigger a collab settlement during ongoing rollover
+    ///
+    /// In this scenario the collab settlement finished before the rollover finished.
+    /// If we trigger a collaborative settlement during rollover, the settlement will have priority
+    /// over the rollover. Upon finishing the rollover we fail because the cfd was already
+    /// settled.
+    #[test]
+    fn given_collab_settlement_then_cannot_finish_rollover() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+
+        let taker_keys = crate::keypair::new(&mut rand::thread_rng());
+        let maker_keys = crate::keypair::new(&mut rand::thread_rng());
```

Comment:
Would be nice to get rid of the keys on this level if we don't need them for anything else beyond the `with_lock` invocation.

Reply:
yeah, but for *most* cases we need them outside because they have to be the same for taker and maker. I think this should go away once we get your changes for a simplified lock in that we don't build collaboratively.